tracer/test/Datadog.Trace.Tests/TraceContextTests.cs

```diff
@@ -195,15 +195,87 @@ public void PartialFlushShouldPropagateMetadata(bool inAASContext)
             tracer.Setup(t => t.Write(It.IsAny<ArraySegment<Span>>()))
                   .Callback<ArraySegment<Span>>(s => spans = s);
 
-            var vars = Environment.GetEnvironmentVariables();
+            SetAASContext(inAASContext);
+            var traceContext = new TraceContext(tracer.Object);
+            traceContext.SetSamplingPriority(SamplingPriorityValues.UserKeep);
+
+            var rootSpan = CreateSpan();
+
+            traceContext.AddSpan(rootSpan);
+
+            for (int i = 0; i < partialFlushThreshold - 1; i++)
+            {
+                var span = CreateSpan();
+                traceContext.AddSpan(span);
+                traceContext.CloseSpan(span);
+            }
+
+            // At this point, only one span is missing to reach the threshold for partial flush
+            spans.Should().BeNull("partial flush should not have been triggered");
+
+            // Closing the root span brings the number of closed spans to the threshold
+            // but a full flush should be triggered rather than a partial, because every span in the trace has been closed
+            traceContext.CloseSpan(rootSpan);
+
+            spans.Value.Should().NotBeNullOrEmpty("a full flush should have been triggered");
+
+            rootSpan.GetMetric(Metrics.SamplingPriority).Should().Be(SamplingPriorityValues.UserKeep, "priority should be assigned to the root span");
+
+            spans.Value.Should().OnlyContain(s => s == rootSpan || s.GetMetric(Metrics.SamplingPriority) == null, "only the root span should have a priority");
```

Comment:
I'm inclined to either remove these assertions if they're tested elsewhere, or to change the name of the test, as these don't have anything to do with AAS

Reply:
Makes sense, hasty copy paste. I've also factorize the AAS decoration checks. Done in 787833a7418157b811f8aeeb747f21bed45d083a